packages/dom/test/serialize-inputs.test.js

```diff
@@ -84,9 +91,20 @@ describe('serializeInputs', () => {
     });
 
     it(`${platform}: serializes checked radio buttons`, () => {
+      expect($('#radio')[0].outerHTML).toContain('checked=""');
       expect($('#radio')[0].checked).toBe(true);
     });
 
+    it(`${platform}: removes checked attr from radio-buttons if present in dom by default but radio is not checked`, () => {
+      // validates `checked=""` is removed from the dom for option1 as it is not checked
+      expect($('#option1')[0].outerHTML).not.toContain('checked=""');
+      expect($('#option1')[0].checked).toBe(false);
+
+      // validates `checked=""` is not removed from the dom for option2 as it is in checked state
+      expect($('#option2')[0].outerHTML).toContain('checked=""');
```

Comment:
In our example, the 2nd anyway shows up as checked. We want to simulate a condition that post this someone selects first radio button and the same is selected in the serialized dom